django/core/management/commands/shell.py

```diff
@@ -111,6 +127,55 @@ def python(self, options):
         # Start the interactive interpreter.
         code.interact(local=imported_objects)
 
+    def get_app_name(self, app_module):
+        rparts = list(reversed(app_module.split(".")))
+        try:
+            try:
+                return rparts[rparts.index(MODELS_MODULE_NAME) + 1]
+            except ValueError:
+                return rparts[1]
+        except IndexError:
+            return app_module
+
+    def get_apps_and_models(self):
+        for app_conf in apps.get_app_configs():
+            if app_conf.models_module:
+                yield app_conf.models_module, app_conf.get_models()
+
+    def get_namespace(self, options, style):
+        """Override this method with import_items in order to import extra_things"""
+
+        imported_objects = {}
+
+        for app_module, app_models in self.get_apps_and_models():
+            if not app_models:
+                continue
+            model_labels = []
+            for model in app_models:
+                if model.__module__:
+                    try:
+                        imported_objects[model.__name__] = import_string(
+                            "%s.%s" % (model.__module__, model.__name__)
+                        )
+                        model_labels.append(model.__name__)
+                    except ImportError as e:
+                        if options.get("traceback"):
+                            traceback.print_exc()
```

Comment:
The `traceback` option is specific to `CommandError`. We should not repurpose it here. 

I also think we don’t need to catch `ImportError`, since it is impossible. As noted above, the `import_string()` call is redundant.